src/test/java/com/fasterxml/jackson/databind/deser/OrderMapEntriesByKeysSerializationFeature4773Test.java

```diff
@@ -75,4 +80,25 @@ void testSerializationWithGenericObjectKeys()
                 "'5':'N_TEXT'}}"), jsonResult);
     }
 
+    @Test
+    void testSerializationSuccessWhenDisabledWithMixedTypes()
```

Comment:
.

Reply:
Right, I don't think we need to support mixed-case with this PR. We can consider follow-up issue if it resurfaces. Javadocs can clarify that feature only checks one of key types to determine Comparability.